diff --git a/r8125-9.011.00/src/r8125.h b/r8125-9.011.00/src/r8125.h
index 2aad6c5..e5e9894 100755
--- a/r8125-9.011.00/src/r8125.h
+++ b/r8125-9.011.00/src/r8125.h
@@ -398,7 +398,7 @@ do { \
 #define RSS_SUFFIX ""
 #endif
 
-#define RTL8125_VERSION "9.011.00" NAPI_SUFFIX DASH_SUFFIX REALWOW_SUFFIX PTP_SUFFIX RSS_SUFFIX
+#define RTL8125_VERSION "9.011.01" NAPI_SUFFIX DASH_SUFFIX REALWOW_SUFFIX PTP_SUFFIX RSS_SUFFIX
 #define MODULENAME "r8125"
 #define PFX MODULENAME ": "
 
@@ -676,11 +676,11 @@ typedef int *napi_budget;
 typedef struct napi_struct *napi_ptr;
 typedef int napi_budget;
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 0)
-#define RTL_NAPI_CONFIG(ndev, priv, function, weight)   netif_napi_add(ndev, &priv->napi, function, weight)
-#else
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,1,0)
 #define RTL_NAPI_CONFIG(ndev, priv, function, weight)   netif_napi_add_weight(ndev, &priv->napi, function, weight)
-#endif
+#else
+#define RTL_NAPI_CONFIG(ndev, priv, function, weight)   netif_napi_add(ndev, &priv->napi, function, weight)
+#endif //LINUX_VERSION_CODE >= KERNEL_VERSION(6,1,0)
 #define RTL_NAPI_QUOTA(budget, ndev)            min(budget, budget)
 #define RTL_GET_PRIV(stuct_ptr, priv_struct)        container_of(stuct_ptr, priv_struct, stuct_ptr)
 #define RTL_GET_NETDEV(priv_ptr)            struct net_device *dev = priv_ptr->dev;
diff --git a/r8125-9.011.00/src/r8125_n.c b/r8125-9.011.00/src/r8125_n.c
index 6ca7201..6079c6c 100755
--- a/r8125-9.011.00/src/r8125_n.c
+++ b/r8125-9.011.00/src/r8125_n.c
@@ -16698,6 +16698,8 @@ rtl8125_rx_interrupt(struct net_device *dev,
 
                 skb->protocol = eth_type_trans(skb, dev);
 
+                total_rx_bytes += skb->len;
+
                 if (skb->pkt_type == PACKET_MULTICAST)
                         total_rx_multicast_packets++;
 
@@ -16707,7 +16709,6 @@ rtl8125_rx_interrupt(struct net_device *dev,
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4,11,0)
                 dev->last_rx = jiffies;
 #endif //LINUX_VERSION_CODE < KERNEL_VERSION(4,11,0)
-                total_rx_bytes += skb->len;
                 total_rx_packets++;
 
 #ifdef ENABLE_PAGE_REUSE
